lib/publiccloud/utils.pm

```diff
@@ -382,9 +382,16 @@ sub prepare_ssh_tunnel {
     assert_script_run("install -o $testapi::username -g users -m 0600 ~/.ssh/* /home/$testapi::username/.ssh/");
 
     # Permit root passwordless login and TCP forwarding over SSH
-    $instance->ssh_assert_script_run('sudo sed -i "s/PermitRootLogin no/PermitRootLogin prohibit-password/g" /etc/ssh/sshd_config');
-    $instance->ssh_assert_script_run('sudo sed -i "/^AllowTcpForwarding/c\AllowTcpForwarding yes" /etc/ssh/sshd_config') if (is_hardened());
+    return if (script_run("sudo sshd -G | grep 'permitrootlogin yes'") == 0 && script_run("sshd -G | grep 'allowtcpforwarding yes'") == 0);
```

Comment:
The early return prevents a lot of things from happening later. Is this intended?

Reply:
🤦 no

Reply:
Fixed. Please check again.